catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -914,65 +917,82 @@ CreateInfo parseAttachments(List<Attachment> contentParts, String transformerPar
+        metacardTypes.stream()
                 .map(metacardType -> metacardType.getAttributeDescriptor(parsedName))
                 .filter(Objects::nonNull)
                 .findFirst()
                 .map(AttributeDescriptor::getType)
-                .map(AttributeType::getAttributeFormat);
-        attributeFormat.ifPresent(attributeFormat1 -> {
-            try {
-                switch (attributeFormat1) {
-                case XML:
-                case GEOMETRY:
-                case STRING:
-                    attributes.add(new AttributeImpl(parsedName, IOUtils.toString(inputStream)));
-                    break;
-                case BOOLEAN:
-                    attributes.add(new AttributeImpl(parsedName, Boolean.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case SHORT:
-                    attributes.add(new AttributeImpl(parsedName, Short.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case LONG:
-                    attributes.add(new AttributeImpl(parsedName, Long.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case INTEGER:
-                    attributes.add(new AttributeImpl(parsedName, Integer.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case FLOAT:
-                    attributes.add(new AttributeImpl(parsedName, Float.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case DOUBLE:
-                    attributes.add(new AttributeImpl(parsedName, Double.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case DATE:
-                    Instant instant = Instant.parse(IOUtils.toString(inputStream));
-                    if (instant == null) {
-                        break;
-                    }
+                .map(AttributeType::getAttributeFormat)
+                .ifPresent((attributeFormat) -> parseAttribute(attributes,
+                        parsedName,
+                        inputStream,
+                        attributeFormat));
+    }
+
+    private void parseAttribute(List<Attribute> attributes, String parsedName,
+            InputStream inputStream, AttributeType.AttributeFormat attributeFormat) {
+        try {
+            // ensure the input stream is no more than MAX_INPUT_SIZE
+            InputStream boundedStream = new BoundedInputStream(inputStream, MAX_INPUT_SIZE + 1);
+            byte[] bytes = IOUtils.toByteArray(boundedStream);
+            if (bytes.length > MAX_INPUT_SIZE) {
+                LOGGER.debug("Attribute length is limited to {} bytes", MAX_INPUT_SIZE);
+                return;
+            }
+
+            if (attributeFormat == OBJECT) {
```

Comment:
This should be moved above consuming the input stream so we don't unnecessarily consume it.